core/arch/arm/kernel/generic_boot.c

```diff
@@ -662,3 +663,20 @@ void generic_boot_init_secondary(unsigned long nsec_entry)
 	init_secondary_helper(nsec_entry);
 }
 #endif
+
+#ifdef CFG_BOOT_SECONDARY_REQUEST
+int generic_smp_boot(size_t core_idx, uint32_t entry)
+{
+	if ((core_idx == 0) || (core_idx >= CFG_TEE_CORE_NB_CORE))
+		return -1;
+
+	/* set secondary cores' NS entry addresses */
+
+	ns_entry_addrs[core_idx] = entry;
```

Comment:
You should vaditate the entry point address.
Verify at least that it is not inside TEE RAM or TA RAM.